homeassistant/components/iammeter/sensor.py

```diff
@@ -0,0 +1,117 @@
+"""Support for iammeter via local API."""
+import asyncio
+from datetime import timedelta
+import logging
+
+import async_timeout
+from iammeter import real_time_api
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_time_interval
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_PORT = 80
+DEFAULT_DEVICE_NAME = "IamMeter"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_DEVICE_NAME): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    }
+)
+
+SCAN_INTERVAL = timedelta(seconds=30)
+PLATFORM_TIMEOUT = 8
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Platform setup."""
+    try:
+        with async_timeout.timeout(PLATFORM_TIMEOUT):
+            api = await real_time_api(config[CONF_HOST], config[CONF_PORT])
+            endpoint = RealTimeDataEndpoint(hass, api)
+            devices = []
+            for sensor, (row, idx, unit) in api.iammeter.sensor_map().items():
+                uid = f"{config[CONF_NAME]}-{api.iammeter.mac}-{api.iammeter.serial_number}-{row}-{idx}"
+                devices.append(
+                    IamMeter(
+                        uid, api.iammeter.serial_number, sensor, unit, config[CONF_NAME]
+                    )
+                )
+            endpoint.sensors = devices
+            async_add_entities(devices)
+            async_track_time_interval(hass, endpoint.async_refresh, SCAN_INTERVAL)
+    except asyncio.TimeoutError:
```

Comment:
Without any logging the user does not know why the setup failed.

Reply:
log info added.